skglm/estimators.py

```diff
@@ -26,6 +29,120 @@
 )
 from skglm.utils import compiled_clone
 from skglm.solvers import cd_solver_path, bcd_solver_path
+from skglm.solvers.cd_solver import cd_solver
+
+
+def glm_fit(X, y, model, datafit, penalty):
+    is_classif = False
+
+    for base in model.__class__.__bases__:
+        if base.__name__ in ["ClassifierMixin", "LinearClassifierMixin"]:
+            is_classif = True
+    if y is None:
+        raise ValueError("requires y to be passed, but the target y is None")
+
+    penalty = penalty if penalty else L1(1.)
+    datafit = datafit if datafit else Quadratic()
+
+    if is_classif:
+        check_classification_targets(y)
+        enc = LabelEncoder()
+        y = enc.fit_transform(y)
+        model.classes_ = enc.classes_
+        n_classes_ = len(model.classes_)
+        sparse_format = "csc" if isinstance(datafit, Logistic) else "csr"
+        is_sparse = issparse(X)
+        if n_classes_ <= 2:
+            y = 2 * y - 1
+        X = check_array(
+            X,
+            accept_sparse=sparse_format,
+            dtype=np.float64)
+
+        y = check_array(y, ensure_2d=False, dtype=X.dtype.type)
+        check_consistent_length(X, y)
+    else:
+        check_X_params = dict(
+            dtype=[np.float64, np.float32], order='F',
+            accept_sparse='csc', copy=model.fit_intercept)
+        check_y_params = dict(ensure_2d=False, order='F')
+
+        X, y = model._validate_data(
+            X, y, validate_separately=(check_X_params, check_y_params))
+        X = check_array(X, 'csc', dtype=[np.float64, np.float32],
+                        order='F', copy=False, accept_large_sparse=False)
+        y = check_array(y, 'csc', dtype=X.dtype.type, order='F', copy=False,
+                        ensure_2d=False)
+    if y.ndim == 2 and y.shape[1] == 1:
+        warnings.warn("DataConversionWarning('A column-vector y"
+                      " was passed when a 1d array was expected")
+        y = y[:, 0]
+    penalty = compiled_clone(penalty)
+    datafit = compiled_clone(datafit, to_float32=X.dtype == np.float32)
+    if issparse(X):
+        datafit.initialize_sparse(X.data, X.indptr, X.indices, y)
+    else:
+        datafit.initialize(X, y)
+
+    if not hasattr(model, "n_features_in_"):
+        model.n_features_in_ = X.shape[1]
+
+    n_samples = X.shape[0]
+    if n_samples != y.shape[0]:
+        raise ValueError("X and y have inconsistent dimensions (%d != %d)"
+                         % (n_samples, y.shape[0]))
+
+    if not model.warm_start or not hasattr(model, "coef_"):
+        model.coef_ = None
+
+    if isinstance(datafit, QuadraticSVC):
+        if issparse:
+            yXT = (X.T).multiply(y)
+        else:
+            yXT = (X * y[:, None]).T
+
+    X_ = yXT if isinstance(datafit, QuadraticSVC) else X
+    # TODO allow warm start
```

Comment:
This can be done L97, with an else following https://github.com/scikit-learn-contrib/skglm/pull/54/files#diff-5a55ebbdbb198f456b4549cc660c2b1e6fc838fd05424e8f4b7775239aed32b9R96